src/Rules/PhpDoc/IncompatibleRequireExtendsTypeTraitRule.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PhpDoc;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InTraitNode;
+use PHPStan\Rules\ClassCaseSensitivityCheck;
+use PHPStan\Rules\ClassNameNodePair;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\VerbosityLevel;
+use function array_merge;
+use function sprintf;
+
+/**
+ * @implements Rule<InTraitNode>
+ */
+class IncompatibleRequireExtendsTypeTraitRule implements Rule
+{
+
+	public function __construct(
+		private ClassCaseSensitivityCheck $classCaseSensitivityCheck,
+		private bool $checkClassCaseSensitivity,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return InTraitNode::class;
```

Comment:
You're right, this one is better served by `Node\Stmt\Trait_` - even when the trait is used multiple times, we just want to report the declaration once.